src/vivarium_gates_mncnh/components/antenatal_care.py

```diff
@@ -0,0 +1,292 @@
+from vivarium.types import ClockTime
+
+from vivarium_gates_mncnh.constants.data_values import (
+    ANC_RATES,
+    COLUMNS,
+    LOW_BIRTH_WEIGHT_THRESHOLD,
+    SIMULATION_EVENT_NAMES,
+    ULTRASOUND_TYPES,
+)
+from vivarium_gates_mncnh.utilities import get_location
+
+
+class TreeMachine(Machine):
+    @property
+    def columns_created(self) -> list[str]:
+        return [self.state_column]
+
+    @property
+    def columns_required(self) -> list[str] | None:
+        return None
+
+    def __init__(self, state_column: str, states: list[State], initial_state: State) -> None:
+        super().__init__(state_column, states)
+        self.initial_state = initial_state
+
+    def setup(self, builder: Builder) -> None:
+        super().setup(builder)
+        self._sim_step_name = builder.time.simulation_event_name()
+
+    def on_initialize_simulants(self, pop_data: SimulantData) -> None:
+        self.population_view.update(
+            pd.Series(
+                self.initial_state.state_id, index=pop_data.index, name=self.state_column
+            ),
+        )
+
+    def on_time_step(self, event: Event) -> None:
+        if self._sim_step_name() == SIMULATION_EVENT_NAMES.PREGNANCY:
+            self.transition(event.index, event.time)
+
+
+class DecisionTreeState(State):
+    def setup(self, builder: Builder) -> None:
+        self._sim_step_name = builder.time.simulation_event_name()
+        self.location = get_location(builder)
+
+    def add_decision(
+        self,
+        output_state: State,
+        decision_function: Callable[[pd.Index], pd.Series],
+    ) -> None:
+        transition = Transition(
+            self, output_state, self.get_probability_function(decision_function)
+        )
+        self.add_transition(transition)
+
+    def get_probability_function(
+        self, decision_function: Callable[[pd.Index], pd.Series]
+    ) -> Callable[[pd.Index], pd.Series]:
+        """We need to check the simulation step name within the probability function, because it will change"""
+
+        def probability_function(index: pd.Index) -> pd.Series:
+            if self._sim_step_name() != SIMULATION_EVENT_NAMES.PREGNANCY:
+                return lambda index: pd.Series(0.0, index=index)
+            return decision_function(index)
+
+        return probability_function
+
+
+class TransientDecisionTreeState(DecisionTreeState, Transient):
+    pass
+
+
+class ANCState(TransientDecisionTreeState):
+    def __init__(self) -> None:
+        super().__init__("attended_antental_care")
+
+    @property
+    def columns_required(self) -> list[str]:
+        return [COLUMNS.ATTENDED_CARE_FACILITY]
+
+    def transition_side_effect(self, index: pd.Index, _event_time: ClockTime) -> None:
+        pop = self.population_view.get(index)
+        pop[COLUMNS.ATTENDED_CARE_FACILITY] = True
+        self.population_view.update(pop)
+
+
+class StandardUltrasound(TransientDecisionTreeState):
```

Comment:
`StandardUltrasound` and `AIAssistedUltrasound` can be combined into a single class. 

```python
class Ultrasound(TransientDecisionTreeState):

    def __init__(self, ultrasound_type: str) -> None:
        super().__init__(f"{ultrasound_type}_ultasound")
        self.ultrasound_type = ultrasound_type

    @property
    def columns_required(self) -> list[str]:
        return [COLUMNS.ULTRASOUND_TYPE]

    def transition_side_effect(self, index: pd.Index, _event_time: ClockTime) -> None:
        pop = self.population_view.get(index)
        pop[COLUMNS.ULTRASOUND_TYPE] = self.ultrasound_type
        self.population_view.update(pop)
```